### PR TITLE
Linear Ops Pruning Support

### PR DESCRIPTION
**Pull request by daniil-lyakhov**
### Changes

Support of the linear operations by pruning algorithm (Mamul, Dense).
Linear layer attributes was added to TFNNCFNetwork
Minor bug fixed in one of the TensorProcessor

### Reason for changes

To extend pruning algo model coverage.

### Related tickets

65689

### Tests
* Unit tests on linear pruning operation
* Linear layer was added to default test pruning model, all pruning tests was adjusted on linear layers
* Mobilenet_v3 unit with linear layer inside se block and simple model with linear layer was added to pruning model analysis test

### CODE REVIEW

**Review by evgeniya-egupova — APPROVED** [review]

File: nncf/tensorflow/pruning/operations.py [hunk 1]
```diff
@@ -90,7 +96,7 @@ class TFFlattenOps(FlattenPruningOp):
 
 @TF_PRUNING_OPERATOR_METATYPES.register('stop_propagation_ops')
 class TFStopMaskForwardPruningOp(StopMaskForwardPruningOp):
-    additional_types = ['Dense', 'MatMul']
```
Comment (vshampor) [inline nncf/tensorflow/pruning/operations.py:93]: 👍

File: nncf/tensorflow/pruning/tensor_processor.py [hunk 2]
```diff
@@ -43,7 +43,7 @@ def assert_allclose(cls, tensors: List[NNCFTensor]) -> None:
 
     @classmethod
     def repeat(cls, tensor: NNCFTensor, repeats: int) -> NNCFTensor:
-        ret_tensor = tf.repeat(tensor, repeats=repeats)
+        ret_tensor = tf.repeat(tensor.tensor, repeats=repeats)
```
Comment (vshampor) [inline nncf/tensorflow/pruning/tensor_processor.py:46]: Why did it work before?

File: tests/torch/pruning/test_model_pruning_analysis.py [hunk 3]
```diff
@@ -170,23 +170,45 @@ def __init__(self, model: Union[Type[torch.nn.Module], Callable[[], torch.nn.Mod
...
+                                                           'TestModelResidualConnection/NNCFConv2d[conv5]/conv2d_0'],
                                   pruned_groups=[['TestModelResidualConnection/NNCFConv2d[conv1]/conv2d_0',
                                                   'TestModelResidualConnection/NNCFConv2d[conv2]/conv2d_0',
                                                   'TestModelResidualConnection/NNCFConv2d[conv3]/conv2d_0']],
                                   pruned_groups_by_node_id=[[1, 2, 4]],
                                   can_prune_after_analysis={0: True, 1: True, 2: True, 3: True, 4: True,
-                                                            5: True, 6: True, 7: False, 8: False, 9: False,
-                                                            10: False, 11: False, 12: False},
+                                                            5: True, 6: True, 7: True, 8: True, 9: True,
+                                                            10: True, 11: True, 12: True},
```
Comment (vshampor) [inline tests/torch/pruning/test_model_pruning_analysis.py:183]: Now the test cases are vulnerable against a bug in the logic when the code would mistakenly report every node is reported as "prunable". Make sure that there are enough test cases where some of the nodes are actually not prunable as the reference results.

File: tests/torch/pruning/test_model_pruning_analysis.py [hunk 4]
```diff
@@ -507,8 +558,8 @@ def __init__(self, model: nn.Module, ref_can_prune: dict):
 MODEL_ANALYSER_TEST_CASES = [
     ModelAnalyserTestStruct(
         model=TestModelResidualConnection,
-        ref_can_prune={0: True, 1: True, 2: True, 3: True, 4: True, 5: True, 6: True, 7: False, 8: False, 9: False,
-                       10: False, 11: False, 12: False}
+        ref_can_prune={0: True, 1: True, 2: True, 3: True, 4: True, 5: True, 6: True, 7: True, 8: True, 9: True,
+                       10: True, 11: True, 12: True}
```
Comment (vshampor) [inline tests/torch/pruning/test_model_pruning_analysis.py:562]: Same comment as above.

**Review by vshampor — CHANGES_REQUESTED** [review]

**Review by alexsu52 — CHANGES_REQUESTED** [review]
Please add support for relu6 and Batchnorm1d operations.

**Comment by daniil-lyakhov** [comment]
> Please add support for relu6 and Batchnorm1d operations.

I've checked MNV3 with RELU6 + batchnorm1d on latest commit. Looks like all types of batchnorms are already supported. Relu6 metatype was really needed for correct pruning.

Please check on MNV3 075

File: nncf/tensorflow/pruning/tensor_processor.py (hunk 2 above)
Comment (daniil-lyakhov) [reply nncf/tensorflow/pruning/tensor_processor.py:46]: I covered this it this pr

File: tests/torch/pruning/test_model_pruning_analysis.py (hunk 3 above)
Comment (daniil-lyakhov) [reply tests/torch/pruning/test_model_pruning_analysis.py:183]: Agee, now count of prunable layers is increased, so I've added test with mobilenet_v3 with mean instead of avgpool layer. Here model analysis should indicate all nodes is unprunalbe. New test is covering thins

File: tests/torch/pruning/test_model_pruning_analysis.py (hunk 4 above)
Comment (daniil-lyakhov) [reply tests/torch/pruning/test_model_pruning_analysis.py:587]: Check comment above

**Comment by daniil-lyakhov** [comment]
> Please add support for relu6 and Batchnorm1d operations.

Batchnorm1d will be covered by #1094

**Review by alexsu52 — APPROVED** [review]
👍

**Review by vshampor — APPROVED** [review]

**Comment by alexsu52** [comment]
@daniil-lyakhov, please, rebase.

**Comment by daniil-lyakhov** [comment]
All updated pruned graph are checked manually

**Comment by daniil-lyakhov** [comment]
Waiting for #1116 merge
